osu.Game/Screens/Ranking/ResultsScreen.cs

```diff
@@ -169,32 +203,82 @@ public override void OnEntering(IScreen last)
 
         public override bool OnExiting(IScreen next)
         {
+            if (statisticsPanel.State.Value == Visibility.Visible)
+            {
+                statisticsPanel.Hide();
+                return true;
+            }
+
             Background.FadeTo(1, 250);
 
             return base.OnExiting(next);
         }
 
-        private class ResultsScrollContainer : OsuScrollContainer
+        private void addScore(ScoreInfo score)
         {
-            private readonly Container content;
+            var panel = scorePanelList.AddScore(score);
 
-            protected override Container<Drawable> Content => content;
+            if (detachedPanel != null)
+                panel.Alpha = 0;
+        }
+
+        private ScorePanel detachedPanel;
 
-            public ResultsScrollContainer()
+        private void onStatisticsStateChanged(ValueChangedEvent<Visibility> state)
+        {
+            if (state.NewValue == Visibility.Visible)
             {
-                base.Content.Add(content = new Container
-                {
-                    RelativeSizeAxes = Axes.X
-                });
+                // Detach the panel in its original location, and move into the desired location in the local container.
+                var expandedPanel = scorePanelList.GetPanelForScore(SelectedScore.Value);
+                var screenSpacePos = expandedPanel.ScreenSpaceDrawQuad.TopLeft;
 
-                RelativeSizeAxes = Axes.Both;
-                ScrollbarVisible = false;
-            }
+                // Detach and move into the local container.
+                scorePanelList.Detach(expandedPanel);
+                detachedPanelContainer.Add(expandedPanel);
+
+                // Move into its original location in the local container.
+                var origLocation = detachedPanelContainer.ToLocalSpace(screenSpacePos);
+                expandedPanel.MoveTo(origLocation);
+                expandedPanel.MoveToX(origLocation.X);
```

Comment:
Does the `MoveToX` need to exist? It seems superfluous, unless I'm missing something...

Reply:
It seems like the X transform that touches property `X` doesn't cancel the movement transform that touches `Position`. But I can improve this one - it previously wasn't extracted into the `origLocation` field.